base/Enums.jl

```diff
@@ -29,7 +29,7 @@ end
 """
     @enum EnumName EnumValue1[=x] EnumValue2[=y]
 
-Create an [`Enum`](:obj:`Enum`) type with name `EnumName` and enum member values of
+Create an `Enum` type with name `EnumName` and enum member values of
```

Comment:
Re: autodetection, I think we can probably add, at some point, a check to Documenter that will suggest non-references that could be references.